src/Adapter/Product/Image/ProductImagePathFactory.php

```diff
@@ -42,32 +48,91 @@ class ProductImagePathFactory
      */
     private $temporaryImgDir;
 
+    /**
+     * @var string
+     */
+    private $basePath;
+
+    /**
+     * @var string
+     */
+    private $pathToBaseDir;
+
+    /**
+     * @var string
+     */
+    private $contextLangIsoCode;
+
     /**
      * @param bool $isLegacyImageMode
+     * @param string $basePath
+     * @param string $relativeImagesDir
      * @param string $temporaryImgDir
+     * @param string $contextLangIsoCode
      */
     public function __construct(
         bool $isLegacyImageMode,
-        string $temporaryImgDir
+        string $basePath,
+        string $relativeImagesDir,
+        string $temporaryImgDir,
+        string $contextLangIsoCode
     ) {
         $this->isLegacyImageMode = $isLegacyImageMode;
-        $this->temporaryImgDir = $temporaryImgDir;
+        // make sure one trailing slash is always there
+        $this->temporaryImgDir = rtrim($temporaryImgDir, '/') . '/';
+        $this->basePath = rtrim($basePath, '/') . '/';
+        $this->pathToBaseDir = $this->basePath . rtrim(ltrim($relativeImagesDir, '/'), '/') . '/';
+        $this->contextLangIsoCode = $contextLangIsoCode;
+    }
+
+    /**
+     * @param string $relativePath
+     *
+     * @return string
+     */
+    public function buildFullPath(string $relativePath): string
+    {
+        return $this->basePath . ltrim($relativePath, '/');
     }
 
     /**
      * @param Image $image
      *
      * @return string
      */
-    public function getBasePath(Image $image): string
+    public function getBaseImagePath(Image $image): string
```

Comment:
MMm, I don't understand why the `Base` suffix, it's simply the image Path
The base is a bit confusing, I would expect the image folder or that image without extension or anything else I don't know

```suggestion
    public function getImagePath(Image $image): string
```